code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -795,3 +795,66 @@
 		R.module.malfhacked = TRUE
 		R.module.rebuild_modules()
 		to_chat(R, "<span class='notice'>New firmware downloaded. Emitter is now online.</span>")
+
+/datum/AI_Module/small/repair_cyborg
+	module_name = "Repair Cyborgs"
+	mod_pick_name = "repair_borg"
+	description = "Causes an electrical surge in the targeted cyborg, rebooting it's subsystems and repairing most of it's subsystems. Requires two uses on a cyborg with broken armor"
+	cost = 20
+	power_type = /datum/action/innate/ai/ranged/repair_cyborg
+	unlock_text = "<span class='notice'>TLB exception on load: Error pointing to address 0000001H, Proceed with execution anywa- SURGE protocalls installed, welcome to open APC!</span>"
+	unlock_sound = 'sound/items/rped.ogg'
+
+/datum/action/innate/ai/ranged/repair_cyborg
+	name = "Repair Cyborg"
+	desc = "Shocks a cyborg back to 'life' after a short delay."
+	button_icon_state = "overload_machine"
+	uses = 2
+	linked_ability_type = /obj/effect/proc_holder/ranged_ai/repair_cyborg
+
+/datum/action/innate/ai/ranged/repair_cyborg/New()
+	..()
+	desc = "[desc] It has [uses] use\s remaining."
```

Comment:
I noticed that this is a re-occurring thing. The desc is also not changed when you use an ability unless you do it yourself.
Might be outside of the scope, but if you want to improve on it: Can you change it to be a bit more sane? So that it auto applies the uses description on `New` if it has uses and also reapplies the text after it gets used?

Reply:
code improvements applied, removed redundant typepaths, unused var, and redundant code in some places